src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -4147,15 +4147,342 @@ static void Main()
         Func<object?> a2 = [return: NotNull] () => null;
     }
 }";
+            // Missing lambda conversion warnings tracked by https://github.com/dotnet/roslyn/issues/56668
             var comp = CreateCompilation(new[] { source, MaybeNullAttributeDefinition, NotNullAttributeDefinition }, parseOptions: TestOptions.RegularPreview);
-            // https://github.com/dotnet/roslyn/issues/52827: Report WRN_NullReferenceReturn for a2, not for a1.
             comp.VerifyDiagnostics(
-                // (8,53): warning CS8603: Possible null reference return.
-                //         Func<object> a1 = [return: MaybeNull] () => null;
-                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "null").WithLocation(8, 53));
+                // (9,52): warning CS8603: Possible null reference return.
+                //         Func<object?> a2 = [return: NotNull] () => null;
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "null").WithLocation(9, 52)
+                );
         }
 
-        [Fact]
+        [Fact, WorkItem(52827, "https://github.com/dotnet/roslyn/issues/52827")]
+        public void LambdaAttributes_NullableAttributes_AnonymousFunctionConversion_Return()
+        {
+            var source =
+@"#nullable enable
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+delegate object D1();
+[return: MaybeNull] delegate object D2();
+[return: NotNull] delegate object? D3();
+
+class Program
+{
+    static void Main()
+    {
+        Func<object> f1 = [return: MaybeNull] () => null; // 1
+        D1 f2 = [return: MaybeNull] () => null; // 2
+        D2 f3 = [return: MaybeNull] () => null;
+        D3 f4 = [return: MaybeNull] () => null; // 3
+
+        Func<object> f5 = [return: NotNull] () => null;
+        D1 f6 = [return: NotNull] () => null;
+        D2 f7 = [return: NotNull] () => null; // 4
+        D3 f8 = [return: NotNull] () => null;
+
+        Func<object?> f9 = [return: NotNull] () => null; // 5
+        D1 f10 = [return: NotNull] object? () => null;
+        D2 f11 = [return: NotNull] object? () => null; // 6
+        D3 f12 = [return: NotNull] object? () => null;
+    }
+}";
+            // Wrong lambda conversion warnings tracked by https://github.com/dotnet/roslyn/issues/56668
+            // We're expecting WRN_NullabilityMismatchInReturnTypeOfTargetDelegate for each commented line
```

Comment:
I'm not 100% sure I understood. Does this mean that this baseline is missing a single diagnostic for each `// n` comment above? And the commented lines are not meant to indicate the entire set of expected diagnostics?

Reply:
That's correct. We should have one WRN_NullabilityMismatchInReturnTypeOfTargetDelegate for each commented line (and only those lines). Other lines have different warnings which are not the focus on this test.

Reply:
Or is it more that the "check nullable attributes in lambda conversion" was implemented, the commented lines were added, then the conversion check was punted out, and the commented lines were intentionally not updated?

Reply:
Your description is also correct (another way to say the same thing).